mm2src/mm2_main/src/lp_swap/max_maker_vol_rpc.rs

```diff
@@ -0,0 +1,147 @@
+    #[display(
+        fmt = "The volume {} of the {} coin less than minimum transaction amount {}",
+        volume,
+        coin,
+        threshold
+    )]
+    VolumeTooLow {
+        coin: String,
+        volume: BigDecimal,
+        threshold: BigDecimal,
+    },
+    #[display(fmt = "No such coin {}", coin)]
+    NoSuchCoin { coin: String },
+    #[display(fmt = "Coin {} is wallet only", coin)]
+    CoinIsWalletOnly { coin: String },
+    #[display(fmt = "Transport error: {}", _0)]
+    Transport(String),
+    #[display(fmt = "Internal error: {}", _0)]
+    InternalError(String),
+}
+
+impl From<CoinFindError> for MaxMakerVolRpcError {
+    fn from(e: CoinFindError) -> Self {
+        match e {
+            CoinFindError::NoSuchCoin { coin } => MaxMakerVolRpcError::NoSuchCoin { coin },
+        }
+    }
+}
+
+impl From<CheckBalanceError> for MaxMakerVolRpcError {
+    fn from(e: CheckBalanceError) -> Self {
+        match e {
+            CheckBalanceError::NotSufficientBalance {
+                coin,
+                available,
+                required,
+                locked_by_swaps,
+            } => MaxMakerVolRpcError::NotSufficientBalance {
+                coin,
+                available,
+                required,
+                locked_by_swaps,
+            },
+            CheckBalanceError::NotSufficientBaseCoinBalance {
+                coin,
+                available,
+                required,
+                locked_by_swaps,
+            } => MaxMakerVolRpcError::NotSufficientBaseCoinBalance {
+                coin,
+                available,
+                required,
+                locked_by_swaps,
+            },
+            CheckBalanceError::VolumeTooLow {
+                coin,
+                volume,
+                threshold,
+            } => MaxMakerVolRpcError::VolumeTooLow {
+                coin,
+                volume,
+                threshold,
+            },
+            CheckBalanceError::Transport(transport) => MaxMakerVolRpcError::Transport(transport),
+            CheckBalanceError::InternalError(internal) => MaxMakerVolRpcError::InternalError(internal),
+        }
+    }
+}
+
+impl HttpStatusCode for MaxMakerVolRpcError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            MaxMakerVolRpcError::NotSufficientBalance { .. }
+            | MaxMakerVolRpcError::NotSufficientBaseCoinBalance { .. }
+            | MaxMakerVolRpcError::VolumeTooLow { .. }
+            | MaxMakerVolRpcError::NoSuchCoin { .. }
+            | MaxMakerVolRpcError::CoinIsWalletOnly { .. } => StatusCode::BAD_REQUEST,
+            MaxMakerVolRpcError::Transport(_) | MaxMakerVolRpcError::InternalError(_) => {
+                StatusCode::INTERNAL_SERVER_ERROR
+            },
+        }
+    }
+}
+
+#[derive(Deserialize)]
+pub struct MaxMakerVolRequest {
+    coin: String,
+}
+
+#[derive(Debug, Serialize)]
+pub struct MaxMakerVolResponse {
+    coin: String,
+    volume: MmNumberMultiRepr,
+}
```

Comment:
We retrieve a lot of additional data during `max_maker_vol` call: balance, locked_amount and fee. I think it might be useful to return all of these to provide more context on why the max volume might differ a lot from available balance.

Reply:
That makes sense! Done